library/alloc/src/macros.rs

```diff
@@ -102,6 +102,10 @@ macro_rules! vec {
 #[macro_export]
 #[stable(feature = "rust1", since = "1.0.0")]
 macro_rules! format {
+    // A faster path for simple format! calls.
+    ("{}", $arg:ident) => {{
```

Comment:
Any reason to make this so restrictive? I think `$arg:expr $(,)?` would catch a lot more and still allow calling `to_string`.

Reply:
https://github.com/rust-lang/rust/pull/76531#discussion_r487375874